src/analytics.js

```diff
@@ -56,3 +57,28 @@ export function triggerAnalyticsEvent(nodeOrDoc, eventType, opt_vars) {
     analytics.triggerEvent(eventType, opt_vars);
   });
 }
+
+/**
+ * Helper method to create analytics element for specific root.
+ * @param {BaseElement} baseInstance
+ * @param {!JSONType} config
+ */
+export function createAnalyticsForExtension(baseInstance, config) {
+  // Note: Require ampdoc include analytics script
+  analyticsForDocOrNull(baseInstance.element).then(analytics => {
+    if (!analytics) {
+      return;
+    }
+    // Create analytics element;
+    const analyticsElem =
+        baseInstance.win.document.createElement('amp-analytics');
+    const scriptElem = createElementWithAttributes(baseInstance.win.document,
+        'script', {
+          'type': 'application/json',
+        });
+    scriptElem.textContent = JSON.stringify(config);
```

Comment:
> What I planned to do is with scoped is: always create new analytics root for scoped element.

Ok, then proceed this way. It sounds good.

> The JSON thing. Sorry I don't quite understand

The code will change slightly. Something like this:
```
let config;
if (this.element.CONFIG) {
  config = this.element.CONFIG;
} else {
  config = JSON.parse(lookupScript().textContent);
}
```